hooks/src/index.js

```diff
@@ -233,7 +233,7 @@ export function useReducer(reducer, initialState, init) {
 					}
 				});
 
-				return shouldUpdate
+				return shouldUpdate || hookState._component.props !== p
```

Comment:
```suggestion
				return shouldUpdate || this.props !== p
```

There are a few of these that seem like they could be referencing `this`, since it's always the current component.